### PR TITLE
CLI ingestion from Jenkins enterprise

### PR DESCRIPTION
**Pull request by rpau**
This PR can only be merged after we can replace the public ingest with the CLI ingest

### CODE REVIEW

**Review by timtebeek — APPROVED** [review]

**Comment by timtebeek** [comment]
We should discuss with the team how some of the removed columns might affect usage at customers. I only have access to the public repos.csv, but the SaaS team might be able to access more repos.csv files. Analysis for public repos.csv below, which we can then repeat as necessary to determine what to keep / lose.

## Style
```bash
cut --delimiter="," --fields 4 repos.csv | sort | uniq -c | sort -h
  10101 
      1 com.netflix.genie.Style
      2 org.openrewrite.java.GoogleJavaFormat
      4 com.netflix.eureka.Style
     23 org.openrewrite.java.SpringFormat
```

## Build action
```bash
cut --delimiter="," --fields 6 repos.csv | sort | uniq -c | sort -h
  10129 
      1 -Dmaven.antrun.skip=true -pl !packaging/pom.xml
      1 maven
      1 -Phadoop_2
```

## Skip reason
```bash
cut --delimiter="," --fields 8 repos.csv | sort | uniq -c | sort -h
   8691 
      1 requires gradle 5.x but doesn't use wrapper
      1 settings.gradle.kts has code that disallows buildscript classpath entries other than those expected which our gradle enterprise integration violates
      7 Build uses Gradle 1.x
     69 Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain (gradlew present but wrapper jar missing)
    147 java.lang.NoClassDefFoundError: com/sun/mirror/apt/AnnotationProcessorFactory
    184 Top-level build tool file is missing
    221 Build uses Gradle 3.x
    373 Build uses Gradle 4.x prior to 4.10 and the artifactory plugin does not support these builds.
    438 Build uses Gradle 2.x
```

## Build tool
```bash
cut --delimiter="," --fields 5 repos.csv | sort | uniq -c | sort -h
      1 buildTool
     92 maven3.3.9
    142 mvnw
    185 
    215 gradle
   3999 gradlew
   5499 maven
```

**Comment by timtebeek** [comment]
Also: this PR currently removes a number of files, the usage and history of which are a bit unknown, as well as if we should replicate the same in the moderne-cli. Could anyone provide some background on these files, and if this should be retained?

## maven/add-mvn-configuration.sh
Appears to write/replace Xmx in `.mvn/jvm.config`. Should we replicate this in the CLI?

## maven/ingest-settings.xml
Commented out presently; would this typically be used by current customers to point to internal repositories?

## groovy/apply-patch.groovy
Applies a patch; is this what's used for the `Validate` button seen on diffs in the SaaS? Should it be retained?

## groovy/run-ingest-pipeline.groovy
Unclear; should it be retained?

File: add-repos.sh [hunk 1]
```diff
@@ -73,35 +73,11 @@ do
     mkdir -p "$clone_dir"
     cd "$clone_dir" || exit
 
-    if ! git clone --depth 1 --no-checkout "https://github.com/$REPO.git"; then
+    if ! git clone --depth 1 --no-checkout "https://$SCM/$REPO.git"; then
```
Comment (timtebeek) [inline add-repos.sh:76]: I can see this was here before already, but I think we give enterprise the option of clone via http(s) or ssh; do we want to keep this as it is here?

**Comment by rpau** [comment]
@timtebeek Very good points!

For the CLI I do believe that we should work on extending the build functionality to support primary the **styles** and the **actions**. The skip reason should be automatically detected by the CLI. This is why we were failing if the gradle version is lower than the expected. I also foresee that those values might have evolved and are outdated. These are not critical functionalities for the COBOL demo, but we should retain them to have a full ingestion story supported in the CLI.

A secondary validation that we should work on is the Java version, but as you noticed I am already reading this value from the CSV. So, not a priority for the COBOL demo. 

The build tool is exactly what we want to avoid. The CLI detects the build tool based on the repository contents and builds according the detection. So, no extra features for this part.

Regarding your second comment:

* **maven/add-mvn-configuration.sh** Is a similar problem than the memory consumption with Gradle? If so, let's do the same.
* **maven/ingest-settings.xml**  I think that this becomes important for the old "publish" function, but it is not required for the CLI, because the repository and credentials need to be defined as a command.
* **groovy/apply-patch.groovy**: The CLI is not supporting fixes, or recipe runs, this is why I have removed this functionality for now. Indeed, the ingestor has nothing to do with patches. Let's ask to the SaaS team about this. We are about the ingest use case. I believe that I have created a tag already for this project to froze the version. We can maintain 2 branches until all the functionality is moved. 
* **groovy/run-ingest-pipeline.groovy** ditto. Let's ask.

**Comment by timtebeek** [comment]
I was wondering if `maven/ingest-settings.xml` might also be needed to build some of the enterprise projects, for instance to have credentials to Maven repositories for internal libraries. That's not an issue currently for the public instance, but I could see that already being a requirement for the SaaS for Java projects.

File: add-repos.sh (hunk 1 above)
Comment (rpau) [reply add-repos.sh:76]: Yup, good point. why not :)

**Comment by rpau** [comment]
@timtebeek you are totally right. Let me figure out how to address all those points :)

**Pull request merged**